app/services/terms_of_service.rb

```diff
@@ -0,0 +1,11 @@
+class TermsOfService
+  def self.tos_accepted?(customer, distributor = nil)
+    return false unless accepted_at = customer&.terms_and_conditions_accepted_at
+
+    if distributor
+      accepted_at > distributor.terms_and_conditions_updated_at
+    else
+      accepted_at > TermsOfServiceFile.updated_at
+    end
```

Comment:
This `if` seems a code smell to me but I can't find a reference right now. There should be two separate methods. One is concerned with the distributor's terms and the other with the platform's terms. They are two separate concerns. It would allow us to pass the distributor to both of them and the code finding the customer record can be DRY in this class and doesn't need to be in the helper.

Anyway, this is still good as is and I don't think we need to change it until we touch it anyway.

Reply:
Ah yeah that's a good catch. I think @sauloperez pointed out a similar antipattern and he knew the name for it. Agree it's not terrible for now though.